### PR TITLE
SDKS-6333: Impressions in memory storage

### PR DESCRIPTION
**Pull request by javrudsky**
# iOS SDK

## What did you accomplish?
- Created in memory impressions storage and tests

### CODE REVIEW

File: SplitTests/Storage/ImpressionsStorageTest.swift [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        TestingHelper.createKeyImpressions(feature: "f1", count: 10).forEach {
+            impressionsStorage.push($0)
+        }
+
+        let count = persistentStorage.storedImpressions.count
+
+        XCTAssertEqual(10, count)
+    }
+
+    func testEnablePersistence() {
```
Comment (gthea) [inline SplitTests/Storage/ImpressionsStorageTest.swift:52]: Maybe add `testDisablePersistence`? if it makes sense

File: Split/Storage/Impressions/ImpressionDao.swift [hunk 2]
```diff
@@ -47,6 +48,33 @@ class CoreDataImpressionDao: BaseCoreDataDao, ImpressionDao {
...
+                        obj.storageId = self.coreDataHelper.generateId()
+                        obj.testName = testName
+                        obj.body = try self.json.encodeToJson(impression)
+                        obj.createdAt = Date().unixTimestamp()
+                        obj.status = StorageRecordStatus.active
+                        // Saving one by one to avoid losing all
+                        // if an error occurs
+                        self.coreDataHelper.save()
+                    } catch {
+                        Logger.e("An error occurred while inserting split in storage: \(error.localizedDescription)")
```
[the remark this answers is not in the document]
Comment (javrudsky) [reply Split/Storage/Impressions/ImpressionDao.swift:71]: 👍  thanks here!

File: Split/Storage/Impressions/ImpressionsStorage.swift [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    init(persistentStorage: PersistentImpressionsStorage,
+         persistenceEnabled: Bool = true) {
+        self.persistentStorage = persistentStorage
+        self.isPersistenceEnabled = Atomic(persistenceEnabled)
+    }
+
+    func enablePersistence(_ enable: Bool) {
+        // Here we should save all impressions
+        isPersistenceEnabled.set(true)
+        persistentStorage.push(impressions: impressions.takeAll())
```
[the remark this answers is not in the document]
Comment (javrudsky) [reply Split/Storage/Impressions/ImpressionsStorage.swift:32]: 🙈 Thanks again!

File: SplitTests/Storage/ImpressionsStorageTest.swift (hunk 1 above)
Comment (javrudsky) [reply SplitTests/Storage/ImpressionsStorageTest.swift:52]: 👍

File: SplitTests/Storage/ImpressionsStorageTest.swift [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+    var persistentStorage: PersistentImpressionsStorageStub!
+
+    override func setUp() {
+        persistentStorage = PersistentImpressionsStorageStub()
+    }
+
+//    func enablePersistence(_ enable: Bool)
+//    func push(_ impression: KeyImpression)
+//    func clearInMemory()
```
[the remark this answers is not in the document]
Comment (javrudsky) [reply SplitTests/Storage/ImpressionsStorageTest.swift:24]: Removed!

**Pull request merged**